src/server/grpc_service.rs

```diff
@@ -796,6 +870,52 @@ fn extract_kv_pairs(res: storage::Result<Vec<storage::Result<storage::KvPair>>>)
     }
 }
 
+fn extract_mvcc_info(key: Key, mvcc: storage::MvccInfo) -> MvccInfo {
+    let mut mvcc_info = MvccInfo::new();
+    if let Some(ref lock) = mvcc.lock {
+        let mut lock_info = LockInfo::new();
+        lock_info.set_primary_lock(lock.primary.clone());
```

Comment:
Do not extract writes and values independently in `extract_info_vectors `, then you can borrow lock directly here and do not need `clone` any more.